src/main/scala/viper/gobra/translator/encodings/typeless/DeferEncoding.scala

```diff
@@ -0,0 +1,78 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) 2011-2020 ETH Zurich.
+
+package viper.gobra.translator.encodings.typeless
+
+import org.bitbucket.inkytonik.kiama.==>
+import viper.gobra.ast.{internal => in}
+import viper.gobra.theory.Addressability
+import viper.gobra.translator.context.Context
+import viper.gobra.translator.encodings.combinators.Encoding
+import viper.gobra.translator.util.{Core, ViperWriter}
+import viper.gobra.translator.util.ViperWriter.{CodeWriter, MemberWriter}
+import viper.silver.{ast => vpr}
+
+class DeferEncoding extends Encoding {
+
+  import viper.gobra.translator.util.ViperWriter.CodeLevel._
+  import viper.gobra.translator.util.ViperWriter.{MemberLevel => ml}
+
+  case class Defer(
+                    activationVar: vpr.LocalVarDecl,
+                    argumentVars: Vector[vpr.LocalVarDecl],
+                    stmt: MemberWriter[vpr.Stmt]
+                  ) extends ViperWriter.CodeCollectible
+
+  override def statement(ctx: Context): in.Stmt ==> CodeWriter[vpr.Stmt] = {
+    case n: in.Defer =>
+      val (pos, info, errT) = n.vprMeta
```

Comment:
I know that this is not super easy for `defer` because its encoding has an impact in the entire control flow of the method, but would it be possible somehow if we had a sketch of the encoding of this feature? If not, we can merge it as it is